### PR TITLE
Simplifying Contribution Docs for SIG CLI

### PR DESCRIPTION
**Pull request by iamNoah1**
Signed-off-by: Noah Ispas (iamNoah1) <noahispas@gmail.com>

This PR is trying to simplify and update the contribution docs as apparently a lot of the content is out of date and does not reflect how the team is working as of today. 

FYI: 
I have removed and updated sections that I thought can be removed or updated. For a lot of things, I didn't just go on and modify without having the team involved. I added some comments here and there with questions and suggestions. 

Summary: 
* Do we still see two tracks of contributions? Guided and Self Service? 
* As there are no issues with labels 'type/*' I would remove the *Picking the right kind of issue* section
* Is the feature request lifecycle up to date? 
* Is the Design Proposal up to date? 
* Section *Merge state meanings* seems not really beneficial for me. Except for the information about when to revisit a PR (6 Months for unmerged PRs). Seems not to be the case, or is it?
* Are all those mentioning aliases used? For me it doesn't look like.

### CODE REVIEW

File: sig-cli/CONTRIBUTING.md
Comment (brianpursley) [inline sig-cli/CONTRIBUTING.md:187]: Not really.

I think this can probably be simplified, and some of the links are no longer relevant.

What would the *actual* process be for a new feature?  Here is my attempt:

1. Open an issue on the k/kubectl repo using "Enhancement Request" (kind/feature label)

2. The monthly bug scrub meeting will triage the issue, which can result in one of the following outcomes:
    a. Issue gets triage/accepted, meaning it is something that the SIG would consider a PR for
    b. Issue is closed, and an explanation is given why the feature is not something that a PR would be accepted for at this time
    c. More information is requested
    d. You are asked to attend a bi-weekly SIG meeting to discuss the feature before moving ahead.

     (2a. Alternatively, you can expedite this process by attending a bi-weekly SIG meeting and put an item on the agenda to discuss the Enhancement Request issue you created)

I don't know if it makes sense to talk about the general PR workflow here or not, because this is not CLI specific, but:

3. You work on the feature and submit a pull request to the k/k repo when you are done.

4. Reviewers will review the PR, either requesting changes or approving and merging it.  This process can take some time, so be patient.

File: sig-cli/CONTRIBUTING.md
Comment (brianpursley) [inline sig-cli/CONTRIBUTING.md:188]: I don't think the design proposal process is what is used anymore.  This is handled more informally during meetings, and then results in a KEP for something that needs a design.

I am not familiar with "design proposals" in the context of Kubernetes, so maybe someone else who has more background can clarify if this should be removed or updated.

File: sig-cli/CONTRIBUTING.md
Comment (brianpursley) [inline sig-cli/CONTRIBUTING.md:267]: I don't think any of these "mention `@kubernetes/sig-cli-pr-reviews`" and others like it are still relevant.

It looks like these might refer to various google groups that aren't really used anymore:
https://groups.google.com/g/kubernetes-sig-cli-pr-reviews

File: sig-cli/CONTRIBUTING.md
Comment (brianpursley) [inline sig-cli/CONTRIBUTING.md:271]: No, I don't think the size of the feature really dictates whether you need e2e tests or not.

When in doubt, add tests.

My best understanding:
* Everything needs unit tests.
* Things that affect the behavior of the CLI input or ouput should have integration tests.
* e2e tests are needed for features that should be tested against a variety of different cluster configurations, or need to have complex assertions of conditions that are not easy to check using a shell script.

I don't know how much in depth we want to go in this section other than just say "you need to add tests for everything you do", and maybe that is good enough for this document.

File: sig-cli/CONTRIBUTING.md
Comment (brianpursley) [inline sig-cli/CONTRIBUTING.md:306]: Yeah, I don't think this is relevant anymore.

File: sig-cli/CONTRIBUTING.md
Comment (brianpursley) [inline sig-cli/CONTRIBUTING.md:381]: As I said before, I don't think any of these `@kubernetes/sig-cli-*` exist anymore, or at least are not used.  Maybe someone else can tell me I'm wrong, but they don't seem to resolve when I type them in a github comment.

What I **do** see are the following, but even then, I'm not sure what these would do:
* `@kubernetes/sig-cli-leads`
* `@kubernetes/sig-cli-kubectl-maintainers`

**Review by brianpursley — CHANGES_REQUESTED** [review]
Thanks for taking this on!  

The document seems to be pretty out-dated and could use some simplification as well, so it is much needed.

I tried to leave some comments that I hope can help, and I'd like to get some review from someone else form the SIG to make sure I'm not telling you something that is wrong.  This document really does need a lot of work to improve it.

My tendency is to want to make things simpler and remove details if they are unnecessary, but I don't want to suggest removing something that might actually be important and I just don't know it.  

The way I look at it, the purpose of this document should be to say:
* Here is how you engage with SIG-CLI.
* Here is how you find something to work on (bug or feature).
* Here is how you get your change reviewed and merged, and set expectations for what that will be like.

So the more directly that can be communicated, the better in my opinion.

---

To answer your summary questions from the issue:

**Do we still see two tracks of contributions? Guided and Self Service?**
My vote is no.  I don't think we draw a distinction between tracks in practice.

**As there are no issues with labels 'type/*' I would remove the Picking the right kind of issue section**
Yes, remove those.  Also the `size/*` labels seem to not be used.

**Is the feature request lifecycle up to date?**
No.  I tried to brainstorm a little bit in the comments, but I don't think that is how it works currently.

**Is the Design Proposal up to date?**
No, same as above.

**Section Merge state meanings seems not really beneficial for me. Except for the information about when to revisit a PR (6 Months for unmerged PRs). Seems not to be the case, or is it?**
Agreed.  I don't really know what that is referring to.

**Are all those mentioning aliases used? For me it doesn't look like.**
Agreed, they are not used anymore and confusing for a new contributor looking for help.  We should remove anything about the mentions, and guide users to slack, the biweekly meetings, or the bug scrub.

File: sig-cli/CONTRIBUTING.md
Comment (iamNoah1) [reply sig-cli/CONTRIBUTING.md:87]: I will remove it :)

**Pull request closed without merging**
